languages/swift/exercises/concept/higher-order-functions/.docs/introduction.md

```diff
@@ -0,0 +1,120 @@
+### Function types
```

Comment:
There is quite a bit of information in the introduction. Perhaps you could try to shorten it a bit? One area you could look at is to cut down on the number of examples (or possibly merging examples). One thing in particular about the examples is that you could often get away with having just one demonstration of a concept. So instead of:

```swift
var stringAndIntToString: (String, Int) -> String
stringAndIntToString = shoppingList
stringAndIntToString("carrots", 3)
// => "You need to buy 3 carrots."
stringAndIntToString = repeater
stringAndIntToString("carrots", 3)
// => "carrotscarrotscarrots"
```

just have

```swift
var stringAndIntToString: (String, Int) -> String
stringAndIntToString = shoppingList
stringAndIntToString("carrots", 3)
// => "You need to buy 3 carrots."
```

Reply:
Done, though I left the second example in the after, to show an example how two different functions of the same type can be assigned to a variable of the same type.

Reply:
👍